atomic_reactor/plugin.py

```diff
@@ -218,30 +217,31 @@ def run(self, keep_going=False):
                 #   AutoRebuildCanceledException was raised here)
                 raise
             except Exception as ex:
-                msg = "plugin '%s' raised an exception: %r" % (plugin_instance.key, ex)
+                msg = "plugin '%s' raised an exception: %r" % (plugin_class.key, ex)
                 logger.debug(traceback.format_exc())
                 if plugin_is_allowed_to_fail or keep_going:
                     logger.warning(msg)
                     logger.info("error is not fatal, continuing...")
                     if not plugin_is_allowed_to_fail:
                         failed_msgs.append(msg)
                 else:
-                    self.on_plugin_failed(plugin_instance.key, ex)
+                    self.on_plugin_failed(plugin_class.key, ex)
                     logger.error(msg)
                     raise PluginFailedException(msg)
 
                 plugin_response = ex
 
             try:
-                finish_time = datetime.datetime.now()
-                duration = finish_time - start_time
-                seconds = duration.total_seconds()
-                logger.debug("plugin '%s' finished in %ds", plugin_name, seconds)
-                self.save_plugin_duration(plugin_instance.key, seconds)
+                if start_time:
+                    finish_time = datetime.datetime.now()
+                    duration = finish_time - start_time
+                    seconds = duration.total_seconds()
+                    logger.debug("plugin '%s' finished in %ds", plugin_name, seconds)
+                    self.save_plugin_duration(plugin_class.key, seconds)
             except Exception:
-                logger.exception("failed to save plugin duration")
+                logger.error("failed to save plugin duration")
```

Comment:
Oops, I didn't change it back. Tests use FakeLogger, which didn't have exception call - fixing it would be a better idea